### ISSUE TITLE
Issue with Meteor 1.4.2 and Moment 2.15.2

### ISSUE THREAD

**Issue by bduff9** [moment/moment#3531]
After reading the blog post about the current recommended version of Meteor, I upgraded yesterday to 1.4.2.  When doing this, I used the flag --all-packages to ensure everything was up to date.  However, the whole process kept failing with a large logging output.  I traced it back and figured out it was moment trying to update to 2.15.2 that was causing the issue.  To fix, I uninstalled moment from my meteor app, then ran the update (which worked perfectly) and then finally tried to add moment back in.  However it failed with the error at the bottom here.  I was able to successfully add moment version 2.15.1, so I am OK for now, but figured you would want to know about this issue.  I am on Windows 10, 64-bit.  If you need any more information from me, please let me know!

```
 => Errors while adding packages:                                                 

While loading package momentjs:moment@2.15.2:                                     
error: Command failed:                                                            
C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x86
_32\dev_bundle\bin\npm.cmd                                                        
rebuild --update-binary                                                           
node-pre-gyp ERR! Tried to download:                                              
https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.0.14/fse-v1.0.14-node-v46-
win32-ia32.tar.gz                                                                 
node-pre-gyp ERR! Pre-built binaries not found for fsevents@1.0.14 and            
node@4.6.1 (node-v46 ABI) (falling back to source compile with node-gyp)          
Traceback (most recent call last):                                                
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\gyp_main.py",                       
line 16, in <module>                                                              
sys.exit(gyp.script_main())                                                       
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\__init__.py",             
line 545, in script_main                                                          
return main(sys.argv[1:])                                                         
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\__init__.py",             
line 538, in main                                                                 
return gyp_main(args)                                                             
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\__init__.py",             
line 523, in gyp_main                                                             
generator.GenerateOutput(flat_list, targets, data, params)                        
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py",       
line 2003, in GenerateOutput                                                      
sln_projects, project_objects, flat=msvs_version.FlatSolution())                  
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py",       
line 1762, in _GatherSolutionFolders                                              
return _DictsToFolders('', root, flat)                                            
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py",       
line 1715, in _DictsToFolders                                                     
for folder, contents in bucket.iteritems():                                       
AttributeError: 'MSVSProject' object has no attribute 'iteritems'                 
gyp ERR! configure error                                                          
gyp ERR! stack Error: `gyp` failed with exit code: 1                              
gyp ERR! stack     at ChildProcess.onCpExit                                       
(C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\lib\configure.js:305:16)                
gyp ERR! stack     at emitTwo (events.js:87:13)                                   
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)                         
gyp ERR! stack     at Process.ChildProcess._handle.onexit                         
(internal/child_process.js:200:12)                                                
gyp ERR! System Windows_NT 10.0.14393                                             
gyp ERR! command                                                                  
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\bin\\node.exe"                                         
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\lib\\node_modules\\node-gyp\\bin\\node-gyp.js"         
"configure" "--fallback-to-build"                                                 
"--module=C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\momentjs_moment\\2
.15.2\\npm\\node_modules\\.temp-96t8ze\\node_modules\\fsevents\\lib\\binding\\Rele
ase\\node-v46-win32-ia32\\fse.node"                                               
"--module_name=fse"                                                               
"--module_path=C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\momentjs_mome
nt\\2.15.2\\npm\\node_modules\\.temp-96t8ze\\node_modules\\fsevents\\lib\\binding\
\Release\\node-v46-win32-ia32"                                                    
gyp ERR! cwd                                                                      
C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.2\npm\node_mod
ules\.temp-96t8ze\node_modules\fsevents                                           
gyp ERR! node -v v4.6.1                                                           
gyp ERR! node-gyp -v v3.4.0                                                       
gyp ERR! not ok                                                                   
node-pre-gyp ERR! build error                                                     
node-pre-gyp ERR! stack Error: Failed to execute                                  
'C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\bin\node.exe                                                      
C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x86
_32\dev_bundle\lib\node_modules\node-gyp\bin\node-gyp.js                          
configure --fallback-to-build                                                     
--module=C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.2\npm
\node_modules\.temp-96t8ze\node_modules\fsevents\lib\binding\Release\node-v46-win3
2-ia32\fse.node                                                                   
--module_name=fse                                                                 
--module_path=C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.
2\npm\node_modules\.temp-96t8ze\node_modules\fsevents\lib\binding\Release\node-v46
-win32-ia32'                                                                      
(1)                                                                               
node-pre-gyp ERR! stack     at ChildProcess.<anonymous>                           
(C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-pre-gyp\lib\util\compile.js:83:29)          
node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)                          
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:172:7)                
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:829:16)      
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit                
(internal/child_process.js:211:5)                                                 
node-pre-gyp ERR! System Windows_NT 10.0.14393                                    
node-pre-gyp ERR! command                                                         
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\bin\\node.exe"                                         
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\lib\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp"    
"install" "--fallback-to-build"                                                   
node-pre-gyp ERR! cwd                                                             
C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.2\npm\node_mod
ules\.temp-96t8ze\node_modules\fsevents                                           
node-pre-gyp ERR! node -v v4.6.1                                                  
node-pre-gyp ERR! node-pre-gyp -v v0.6.30                                         
node-pre-gyp ERR! not ok                                                          

npm ERR! Windows_NT 10.0.14393                                                    
npm ERR! argv                                                                     
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\bin\\node.exe"                                         
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\lib\\node_modules\\npm\\bin\\npm-cli.js"               
"rebuild" "--update-binary"                                                       
npm ERR! node v4.6.1                                                              
npm ERR! npm  v3.10.9                                                             
npm ERR! code ELIFECYCLE                                                          
npm ERR! fsevents@1.0.14 install: `node-pre-gyp install --fallback-to-build`      
npm ERR! Exit status 1                                                            
npm ERR!                                                                          
npm ERR! Failed at the fsevents@1.0.14 install script 'node-pre-gyp install       
--fallback-to-build'.                                                             
npm ERR! Make sure you have the latest version of node.js and npm installed.      
npm ERR! If you do, this is most likely a problem with the fsevents package,      
npm ERR! not with npm itself.                                                     
npm ERR! Tell the author that this fails on your system:                          
npm ERR!     node-pre-gyp install --fallback-to-build                             
npm ERR! You can get information on how to open an issue for this project         
with:                                                                             
npm ERR!     npm bugs fsevents                                                    
npm ERR! Or if that isn't available, you can get their info via:                  
npm ERR!     npm owner ls fsevents                                                
npm ERR! There is likely additional logging output above.                         

npm ERR! Please include the following file with any support request:              
npm ERR!                                                                          
C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.2\npm\node_mod
ules\.temp-96t8ze\npm-debug.log                                                   
node-pre-gyp ERR! Tried to download:                                              
https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.0.14/fse-v1.0.14-node-v46-
win32-ia32.tar.gz                                                                 
node-pre-gyp ERR! Pre-built binaries not found for fsevents@1.0.14 and            
node@4.6.1 (node-v46 ABI) (falling back to source compile with node-gyp)          
Traceback (most recent call last):                                                
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\gyp_main.py",                       
line 16, in <module>                                                              
sys.exit(gyp.script_main())                                                       
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\__init__.py",             
line 545, in script_main                                                          
return main(sys.argv[1:])                                                         
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\__init__.py",             
line 538, in main                                                                 
return gyp_main(args)                                                             
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\__init__.py",             
line 523, in gyp_main                                                             
generator.GenerateOutput(flat_list, targets, data, params)                        
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py",       
line 2003, in GenerateOutput                                                      
sln_projects, project_objects, flat=msvs_version.FlatSolution())                  
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py",       
line 1762, in _GatherSolutionFolders                                              
return _DictsToFolders('', root, flat)                                            
File                                                                              
"C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py",       
line 1715, in _DictsToFolders                                                     
for folder, contents in bucket.iteritems():                                       
AttributeError: 'MSVSProject' object has no attribute 'iteritems'                 
gyp ERR! configure error                                                          
gyp ERR! stack Error: `gyp` failed with exit code: 1                              
gyp ERR! stack     at ChildProcess.onCpExit                                       
(C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-gyp\lib\configure.js:305:16)                
gyp ERR! stack     at emitTwo (events.js:87:13)                                   
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)                         
gyp ERR! stack     at Process.ChildProcess._handle.onexit                         
(internal/child_process.js:200:12)                                                
gyp ERR! System Windows_NT 10.0.14393                                             
gyp ERR! command                                                                  
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\bin\\node.exe"                                         
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\lib\\node_modules\\node-gyp\\bin\\node-gyp.js"         
"configure" "--fallback-to-build"                                                 
"--module=C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\momentjs_moment\\2
.15.2\\npm\\node_modules\\.temp-96t8ze\\node_modules\\fsevents\\lib\\binding\\Rele
ase\\node-v46-win32-ia32\\fse.node"                                               
"--module_name=fse"                                                               
"--module_path=C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\momentjs_mome
nt\\2.15.2\\npm\\node_modules\\.temp-96t8ze\\node_modules\\fsevents\\lib\\binding\
\Release\\node-v46-win32-ia32"                                                    
gyp ERR! cwd                                                                      
C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.2\npm\node_mod
ules\.temp-96t8ze\node_modules\fsevents                                           
gyp ERR! node -v v4.6.1                                                           
gyp ERR! node-gyp -v v3.4.0                                                       
gyp ERR! not ok                                                                   
node-pre-gyp ERR! build error                                                     
node-pre-gyp ERR! stack Error: Failed to execute                                  
'C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\bin\node.exe                                                      
C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x86
_32\dev_bundle\lib\node_modules\node-gyp\bin\node-gyp.js                          
configure --fallback-to-build                                                     
--module=C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.2\npm
\node_modules\.temp-96t8ze\node_modules\fsevents\lib\binding\Release\node-v46-win3
2-ia32\fse.node                                                                   
--module_name=fse                                                                 
--module_path=C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.
2\npm\node_modules\.temp-96t8ze\node_modules\fsevents\lib\binding\Release\node-v46
-win32-ia32'                                                                      
(1)                                                                               
node-pre-gyp ERR! stack     at ChildProcess.<anonymous>                           
(C:\Users\Duffey\AppData\Local\.meteor\packages\meteor-tool\1.4.2\mt-os.windows.x8
6_32\dev_bundle\lib\node_modules\node-pre-gyp\lib\util\compile.js:83:29)          
node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)                          
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:172:7)                
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:829:16)      
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit                
(internal/child_process.js:211:5)                                                 
node-pre-gyp ERR! System Windows_NT 10.0.14393                                    
node-pre-gyp ERR! command                                                         
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\bin\\node.exe"                                         
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\lib\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp"    
"install" "--fallback-to-build"                                                   
node-pre-gyp ERR! cwd                                                             
C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.2\npm\node_mod
ules\.temp-96t8ze\node_modules\fsevents                                           
node-pre-gyp ERR! node -v v4.6.1                                                  
node-pre-gyp ERR! node-pre-gyp -v v0.6.30                                         
node-pre-gyp ERR! not ok                                                          

npm ERR! Windows_NT 10.0.14393                                                    
npm ERR! argv                                                                     
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\bin\\node.exe"                                         
"C:\\Users\\Duffey\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2\\mt-os.w
indows.x86_32\\dev_bundle\\lib\\node_modules\\npm\\bin\\npm-cli.js"               
"rebuild" "--update-binary"                                                       
npm ERR! node v4.6.1                                                              
npm ERR! npm  v3.10.9                                                             
npm ERR! code ELIFECYCLE                                                          
npm ERR! fsevents@1.0.14 install: `node-pre-gyp install --fallback-to-build`      
npm ERR! Exit status 1                                                            
npm ERR!                                                                          
npm ERR! Failed at the fsevents@1.0.14 install script 'node-pre-gyp install       
--fallback-to-build'.                                                             
npm ERR! Make sure you have the latest version of node.js and npm installed.      
npm ERR! If you do, this is most likely a problem with the fsevents package,      
npm ERR! not with npm itself.                                                     
npm ERR! Tell the author that this fails on your system:                          
npm ERR!     node-pre-gyp install --fallback-to-build                             
npm ERR! You can get information on how to open an issue for this project         
with:                                                                             
npm ERR!     npm bugs fsevents                                                    
npm ERR! Or if that isn't available, you can get their info via:                  
npm ERR!     npm owner ls fsevents                                                
npm ERR! There is likely additional logging output above.                         

npm ERR! Please include the following file with any support request:              
npm ERR!                                                                          
C:\Users\Duffey\AppData\Local\.meteor\packages\momentjs_moment\2.15.2\npm\node_mod
ules\.temp-96t8ze\npm-debug.log                                                   
```

**Comment by tcastelli** [comment]
Also with the same problem, seems to be happening to mostly all users without macOs (no fs_events)
https://github.com/meteor/meteor/issues/7958

**Comment by glasser** [comment]
It looks like something changed in Meteor 1.4.2 to include `./node_modules` when you run `meteor publish`?  @benjamn

**Comment by benjamn** [comment]
Correct, local `node_modules` directories in package directories are now included by `meteor publish`, as they should be, since they are importable by the package.

It appears that [this Grunt script](https://github.com/moment/moment/blob/c790f4fe0df670eb36903476b42c7a231a07daaa/Gruntfile.js#L168) should be changed to copy [moment.js](https://github.com/moment/moment/blob/develop/moment.js) into https://github.com/moment/moment/tree/develop/meteor, so that `meteor publish` can run within the `meteor` directory, rather than copying [meteor/package.js](https://github.com/moment/moment/blob/develop/meteor/package.js) into the root directory and running `meteor publish` there.

**Comment by tcastelli** [comment]
Would it make sense to strip devDependencies that are not imported on the packages before publishing?

**Comment by yohio** [comment]
I'm getting the same thing on a fresh meteor create project.
not sure how to get passed this issue

**Comment by marcosncom** [comment]
I had issues deploying an app with meteor bundle using this version, something related to fibers 1.0.14... jumped over to mrt:moment, using an old version of 2.8 of moment but it works and uses fibers 1.0.15.

**Comment by bduff9** [comment]
@yahio I'm not sure if you read my post but I personally was successful installing one version back (2.15.1).  You might want to give it a shot to see if it works for you too with this command:

`meteor add momentjs:moment@=2.15.1`

**Comment by yohio** [comment]
@bduff9 Thank you! I missed that part on your post, I was more focusing on the error log and saw that it matched mine almost identically. 

Running the specific version worked for me, I did try doing that before but my syntax was wrong because I wrote the same without the equal sign and it didn't work for a reason that is now clear to me.

Thanks!

**Comment by pldrake26** [comment]
I am having the same issue.  I removed moment and finished Meteor update then added moment back at version 2.15.1 which worked for me also.  Thanks for the work around.

**Comment by bduff9** [comment]
@yohio @pldrake26 Great!  Glad it worked for you guys until an official fix comes out

**Comment by bduff9** [comment]
@ichernev Just FYI, you closed this issue but the public releases are still broken as they are at 2.15.2 on npm and atmosphere, not 2.15.3.  Not sure if that was an accident or not, but its misleading to those of us following this issue.

**Comment by abernix** [comment]
This is fixed with the publishing of `momentjs:moment` 2.16.0 [to Atmosphere](https://atmospherejs.com/momentjs/moment).

Meteor users should no longer need to pin their version to 2.15.1 and can fix this problem if they are using moment 2.15.2:

    meteor add momentjs:moment@2.16.0
